### PR TITLE
Added dev docker compose for dev database

### PR DESCRIPTION
**Pull request by Bikram-ghuku**
# Description

Added development docker compose to spin up a PostgreSQL database for the Go backend to connect to.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Run the docker image with docker compose command and test with go run .
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,9 +51,10 @@ IQPS was originally created by [Shubham Mishra](https://github.com/grapheo12) in
 
 ## Development
 1. Clone this repository.
-2. Start the backend by running `go run .` in the `backend/` directory.
-3. Set up the frontend by running `pnpm install` and then `pnpm start` in the `frontend/` directory.
-4. Profit.
+2. Start the Database for backend by running `sudo docker compose -f docker-compose.dev.yaml up -d` in the `backend/` directory.
+3. Start the backend by running `go run .` in the `backend/` directory.
```
Comment (harshkhandeparkar) [inline README.md:55]: Can we add either:
1. A line here that tells the user to set the database host etc to `locahost`, or
2. A `DEV` environment variable which does the same if true.

**Review by harshkhandeparkar — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Bikram-ghuku) [reply README.md]: Made a new commit for the updated README

File: README.md [hunk 2]
```diff
@@ -51,7 +51,12 @@ IQPS was originally created by [Shubham Mishra](https://github.com/grapheo12) in
 
 ## Development
 1. Clone this repository.
-2. Start the backend by running `go run .` in the `backend/` directory.
+2. For starting the backend:
+    - Change directory to backend `cd backend`
+    - Make the env file by copying the template: `cp .env.template .env`
+    - Fill the env variable and set `DB_HOST=localhost` for running locally for development
```
Comment (harshkhandeparkar) [inline README.md:57]: Port also has to be set :)

Comment (harshkhandeparkar) [reply README.md:57]: And all the other variables as well lol

Comment (harshkhandeparkar) [reply README.md:57]: There is already a separate section for env vars. Perhaps you can update it?

**Review by harshkhandeparkar — APPROVED** [review]

File: backend/docker-compose.dev.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+services:
+  database-dev:
+    image: postgres
+    restart: always
+    environment:
+      - POSTGRES_USER=${DB_USER}
+      - POSTGRES_PASSWORD=${DB_PASSWORD}
+      - POSTGRES_DB=${DB_NAME}
```
Comment (shikharish) [inline backend/docker-compose.dev.yaml:8]: You need to add `PGPORT` and `PGHOST` as environment variables here, as such:
```
      - PGPORT=${DB_PORT}
      - PGHOST=${DB_HOST}
```

File: README.md [hunk 4]
```diff
@@ -51,7 +51,12 @@ IQPS was originally created by [Shubham Mishra](https://github.com/grapheo12) in
 
 ## Development
 1. Clone this repository.
-2. Start the backend by running `go run .` in the `backend/` directory.
+2. For starting the backend:
+    - Change directory to backend `cd backend`
+    - Make the env file by copying the template: `cp .env.template .env`
+    - Fill the env variable and set `DB_HOST=localhost` and `DB_PORT=5432` for running locally for development
+    - Start the DB by running `sudo docker compose -f docker-compose.dev.yaml up -d`
```
Comment (shikharish) [inline README.md:58]: no need for sudo here.

File: README.md [hunk 5]
```diff
@@ -51,7 +51,12 @@ IQPS was originally created by [Shubham Mishra](https://github.com/grapheo12) in
 
 ## Development
 1. Clone this repository.
-2. Start the backend by running `go run .` in the `backend/` directory.
+2. For starting the backend:
+    - Change directory to backend `cd backend`
+    - Make the env file by copying the template: `cp .env.template .env`
+    - Fill the env variable and set `DB_HOST=localhost` and `DB_PORT=5432` for running locally for development
```
Comment (shikharish) [inline README.md:57]: DB_HOST will be localhost but let the developer set the port as per their choice.

**Comment by harshkhandeparkar** [comment]
@shikharish merge if you approve.

**Review by shikharish — APPROVED** [review]

**Comment by shikharish** [comment]
Thanks for your contribution!

**Pull request merged**